integration-cli/docker_cli_wait_test.go

```diff
@@ -120,3 +121,44 @@ func TestWaitBlockedExitRandom(t *testing.T) {
 
 	logDone("wait - blocking wait with random exit code")
 }
+
+func TestWaitStartKillContainerRestartDaemonWaitContainer(t *testing.T) {
```

Comment:
haha no worries it's not easy. If you need to spawn a new daemon, maybe i'd put it in docker_cli_daemon_test.go and rename it so that it starts with TestDaemon.